06_puppy/sirigithub/mapstore.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+)
+
+type MapStore map[int]Puppy
+
+func NewMapStore() MapStore {
+	return MapStore{}
+}
+
+// CreatePuppy takes a  user provided puppy, and creates a new Puppy in the store
+// returns the ID of the new Puppy.
+func (m MapStore) CreatePuppy(puppy *Puppy) int {
+	puppy.ID = len(m)
```

Comment:
This will not work as a way of generating IDs. If you add two puppies, delete the first and then add a third puppy, it will get the same id as the second one. You will need to maintain a sequence number yourself and use it to provide the next id:
```
type MapStore struct {
    store  map[int]Puppy
    nextID int
}
```

This also modifies the puppy passed to it. This is an ok decision to make, but it should be documented in the docstring. It is important that a caller know the value they pass is mutated by a function.

Reply:
@camh-anz thanks a lot for spotting the bug, i totally missed that. Updated logic to maintain a sequence that allocates the next ID